front/src/modules/rollingStock/components/RollingStockEditor/RollingStockEditorFormHelpers.tsx

```diff
@@ -95,30 +103,66 @@ export const RollingStockEditorMetadataForm = ({
 };
 
 type RollingStockEditorParameterFormProps = {
-  optionValue: string;
   rollingStockValues: RollingStockParametersValues;
-  setOptionValue: (option: React.SetStateAction<string>) => void;
   setRollingStockValues: (
     rollingStockValue: React.SetStateAction<RollingStockParametersValues>
   ) => void;
 };
 
 // TODO: make the conditional return clearer
 const RollingStockEditorParameterFormColumn = ({
-  optionValue,
   rollingStockValues,
-  setOptionValue,
   setRollingStockValues,
   propertiesList,
 }: RollingStockEditorParameterFormProps & {
   propertiesList: SchemaProperty[];
 }) => {
   const { t } = useTranslation(['rollingstock', 'translation']);
+  const dispatch = useAppDispatch();
+
+  const isConversionPossible = (previousUnit: string, newUnit: string) => {
+    if (isConversionWithTon(previousUnit, newUnit)) {
+      if (newUnit.slice(-1) === 't') {
+        newUnit = newUnit.slice(0, -2);
+      } else {
+        previousUnit = previousUnit.slice(0, -2);
+      }
+    }
+    if (previousUnit === newUnit) return true;
+    return (
+      previousUnit in conversionFactorsSchema && newUnit in conversionFactorsSchema[previousUnit]
+    );
+  };
```

Comment:
- Simplify the logic by reducing the number of conditional branches. Use string.endsWith for clearer checks.
- Make the function cleaner and easier to understand by introducing local variables for modifications.
-  I think moving this function to utils/rolling_stock/helpers/utils is a good idea.
- Add unit tests: successful/unsuccessful conversions, conversions involving tons, cases with identical units, etc...
```suggestion
  const isConversionPossible = (previousUnit: Unit, newUnit: Unit) => {
  let localPreviousUnit = previousUnit;
  let localNewUnit = newUnit;

  if (isConversionWithTon(previousUnit, newUnit)) {
    localNewUnit = newUnit.endsWith('t') ? newUnit.slice(0, -2) : newUnit;
    localPreviousUnit = previousUnit.endsWith('t') ? previousUnit.slice(0, -2) : previousUnit;
  }

  if (localPreviousUnit === localNewUnit) return true;

  return (
    localPreviousUnit in conversionFactorsSchema &&
    localNewUnit in conversionFactorsSchema[localPreviousUnit]
  );
};
```

Reply:
Thank you for the suggestion, forgot about the endsWith() method !

I find it a bit repetitive to reassign newUnit to localNewUnit if condition is false soI did it like below if it's ok for you : 

```suggestion
  const isConversionPossible = (previousUnit: string, newUnit: string) => {
    let localPreviousUnit = previousUnit;
    let localNewUnit = newUnit;

    if (isConversionWithTon(previousUnit, newUnit)) {
      if (newUnit.endsWith('t')) {
        localNewUnit = newUnit.slice(0, -2);
      } else {
        localPreviousUnit = previousUnit.slice(0, -2);
      }
    }
    if (localPreviousUnit === localNewUnit) return true;
    return (
      localPreviousUnit in conversionFactorsSchema &&
      localNewUnit in conversionFactorsSchema[localPreviousUnit]
    );
  };
```

Reply:
Ok, I think that it can be better if we merge the two return, no need to return.

```
    return (
      (localPreviousUnit === localNewUnit) ||
      (localPreviousUnit in conversionFactorsSchema &&
      localNewUnit in conversionFactorsSchema[localPreviousUnit])
    );
```

Reply:
Good point, added it with an `&&` instead on an `||` because we need all conditions together